README.md

```diff
@@ -604,31 +604,25 @@ This alias would search 'up' from your current directory in order to detect a `.
 
 ##### Calling `nvm use` automatically in a directory with a `.nvmrc` file
 
-Put this into your `$HOME/.zshrc` to call `nvm use` automatically whenever you enter a directory that contains an
-`.nvmrc` file with a string telling nvm which node to `use`:
-
+Put this into your `$HOME/.zshrc` after nvm initialization to call `nvm use` automatically.
+Usage: follow the [simple official guide to create a .nvmrc file](https://github.com/nvm-sh/nvm#nvmrc) containing the node version number in the directory you want to use.
 ```zsh
 # place this after nvm initialization!
 autoload -U add-zsh-hook
 load-nvmrc() {
-  local node_version="$(nvm version)"
-  local nvmrc_path="$(nvm_find_nvmrc)"
-
-  if [ -n "$nvmrc_path" ]; then
-    local nvmrc_node_version=$(nvm version "$(cat "${nvmrc_path}")")
+  local nvmrc_path=".nvmrc"
 
-    if [ "$nvmrc_node_version" = "N/A" ]; then
-      nvm install
-    elif [ "$nvmrc_node_version" != "$node_version" ]; then
-      nvm use
+  if [ -f "$nvmrc_path" ]; then
+    # check if current node version is different from that in the config file
+    if [[ "$(which node)" != *"v$(cat "${nvmrc_path}")"* ]]; then
```

Comment:
why would `which node` print out a version number?

also, this won't work because `.nvmrc` can contain `node` or `v14` or `14` or `14.2` or `lts/*` or `lts/erbium` or any user-defined alias.

Reply:
I could use `node --version` instead of `which node` if the aliased path could be something like `lts/*` :).

The other case is tricky. Maybe I could add in "Usage" that this **additional** faster approach would only strictly work for the format like `14.x.y` or `14` (without v). I totally see your pain in balancing between the flexibility of `.nvmrc` and automation like this.

Reply:
There's a reason that the current code uses `nvm_version` - it's because that's literally the only way to ensure that you're supporting everything `.nvmrc` supports.

It is entirely irrelevant if an approach is fast but incorrect; the slowest possible thing is a broken thing, so if it's incorrect, "faster" can't possibly apply.

Reply:
Ok, I think if we ever want to speed up script like this, we might have to rewrite `nvm_version` in a compiled language like c. Feel free to close this PR. Thanks for reviewing it! (Also I figured the manual call to the load was likely to check the node version upon launching a new terminal.)

Reply:
If there’s any way to speed it up in posix, I’d be happy to review that PR. If nvm was ever written in another language, it’d definitely never be c :-)

Reply:
Just curious, why not :-)

Reply:
Because C code is rife with memory bugs and security vulnerabilities. If i was going for a drastic rewrite, and JS was unavailable, I’d use rust.